### PR TITLE
implemented additional logging

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description:

Add logging to marketplace services:
- Added logging to all marketplace services
- Redirected `gsutil cp` commands output from console to log file only 

## Log file example:

[Prepare Content Packs For Testing.log](https://github.com/demisto/content/files/5470616/Prepare.Content.Packs.For.Testing.log)

## Console output example:
[Prepare Content Packs For Testing Console.log](https://github.com/demisto/content/files/5470619/Prepare.Content.Packs.For.Testing.Console.log)

### CODE REVIEW

File: Tests/scripts/prepare_content_packs_for_testing.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ TARGET_PATH="$BUILD_BUCKET_PATH/content/packs"
 PACKS_FULL_TARGET_PATH="$GCS_BUILD_BUCKET/$TARGET_PATH"
 BUCKET_FULL_TARGET_PATH="$GCS_BUILD_BUCKET/$BUILD_BUCKET_PATH"
 echo "Copying master files at: $SOURCE_PATH to target path: gs://$PACKS_FULL_TARGET_PATH ..."
-gsutil -m cp -r "gs://$GCS_MARKET_BUCKET/$SOURCE_PATH" "gs://$PACKS_FULL_TARGET_PATH"
+gsutil -m cp -r "gs://$GCS_MARKET_BUCKET/$SOURCE_PATH" "gs://$PACKS_FULL_TARGET_PATH" > "$CIRCLE_ARTIFACTS/logs/Prepare Content Packs For Testing.log" 2>&1
```
Comment (yaakovi) [inline Tests/scripts/prepare_content_packs_for_testing.sh:33]: are you sure you want to remove this from the circle log and only save to file?
if not, use `tee` please

File: Tests/Marketplace/upload_packs_private.py [hunk 2]
```diff
@@ -234,25 +236,25 @@ def clean_non_existing_packs(index_folder_path, private_packs, storage_bucket):
...
                 invalid_pack_gcs_path = os.path.join(GCPConfig.STORAGE_BASE_PATH, invalid_pack_name, "")  # by design
 
                 for invalid_blob in [b for b in storage_bucket.list_blobs(prefix=invalid_pack_gcs_path)]:
-                    print_warning(f"Deleted invalid {invalid_pack_name} pack under url {invalid_blob.public_url}")
+                    logging.warning(f"Deleted invalid {invalid_pack_name} pack under url {invalid_blob.public_url}")
                     invalid_blob.delete()  # delete invalid pack in gcs
-        except Exception as e:
-            print_error(f"Failed to cleanup non existing packs. Additional info:\n {e}")
+        except Exception:
+            logging.exception("Failed to cleanup non existing packs.")
```
Comment (yaakovi) [inline Tests/Marketplace/upload_packs_private.py:254]: the exception is very broad, are you sure you want to remove the print of it?
you can add it in debug.

relevant for all exceptions

File: Tests/Marketplace/upload_packs_private.py [hunk 3]
```diff
@@ -652,28 +653,22 @@ def print_packs_summary(packs_list):
...
 
-    print("\n")
-    print("------------------------------------------ Packs Upload Summary ------------------------------------------")
-    print(f"Total number of packs: {len(packs_list)}")
-    print("----------------------------------------------------------------------------------------------------------")
-
+    logging.info(
+        f"""------------------------------------------ Packs Upload Summary ------------------------------------------
+        Total number of packs: {len(packs_list)}
+        ----------------------------------------------------------------------------------------------------------""")
```
Comment (yaakovi) [inline Tests/Marketplace/upload_packs_private.py:659]: the extra spaces at the beginning of the lines will mess the formatting.

**Review by yaakovi — COMMENTED** [review]
🆒

File: Tests/scripts/prepare_content_packs_for_testing.sh (hunk 1 above)
Comment (hod-alpert) [reply Tests/scripts/prepare_content_packs_for_testing.sh:33]: Yes, I'm sure.this is pure debug prints - 99% of the times they are not interesting at all, and they make this step console pass the 40,000 chars limit which after that the logs colors are disabled.

File: Tests/Marketplace/upload_packs_private.py (hunk 2 above)
Comment (hod-alpert) [reply Tests/Marketplace/upload_packs_private.py:254]: When you use logging exception in an exception handling context it automatically print the trace and the message of the exception free of charge with ERROR level. 
There is no need to add the print as it's automatically printed

image

File: Tests/Marketplace/upload_packs_private.py (hunk 3 above)
Comment (hod-alpert) [reply Tests/Marketplace/upload_packs_private.py:659]: image

**Review by yaakovi — APPROVED** [review]

**Comment by Itay4** [comment]
build fails on non related tests
merging

**Pull request merged**